app/src/test/java/com/linagora/calendar/app/TWPSyncSettingsIntegrationTest.java

```diff
@@ -0,0 +1,199 @@
+import com.google.inject.multibindings.Multibinder;
+import com.linagora.calendar.app.modules.CalendarDataProbe;
+import com.linagora.calendar.dav.DavModuleTestHelper;
+import com.linagora.calendar.dav.Fixture;
+import com.linagora.calendar.restapi.RestApiServerProbe;
+
+import io.restassured.RestAssured;
+import io.restassured.authentication.PreemptiveBasicAuthScheme;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.parsing.Parser;
+import io.restassured.specification.RequestSpecification;
+import reactor.core.publisher.Mono;
+import reactor.rabbitmq.OutboundMessage;
+
+class TWPSyncSettingsIntegrationTest {
+    private static final Domain DOMAIN = Domain.of("domain.tld");
+    private static final String PASSWORD = "secret";
+    private static final Username USERNAME = Username.of("bob@domain.tld");
+    private static final String LANGUAGE_KEY = "language";
+    private static final String LANGUAGE_FR = "fr";
+    private static final String LANGUAGE_EN = "en";
+    private static final String EXCHANGE_NAME = "settings";
+    private static final String ROUTING_KEY = "user.settings.updated";
+    private static final Username NON_EXISTING_USER = Username.of("nonExisting@domain.tld");
+
+    @RegisterExtension
+    @Order(1)
+    private static RabbitMQExtension rabbitMQExtension = RabbitMQExtension.singletonRabbitMQ()
+        .isolationPolicy(WEAK);
+
+    @RegisterExtension
+    @Order(2)
+    static TwakeCalendarExtension twakeCalendarExtension = new TwakeCalendarExtension(
+        TwakeCalendarConfiguration.builder()
+            .configurationFromClasspath()
+            .userChoice(TwakeCalendarConfiguration.UserChoice.MEMORY)
+            .dbChoice(TwakeCalendarConfiguration.DbChoice.MEMORY),
+        DavModuleTestHelper.RABBITMQ_MODULE.apply(rabbitMQExtension),
+        DavModuleTestHelper.BY_PASS_MODULE,
+        AppTestHelper.OIDC_BY_PASS_MODULE,
+        binder -> Multibinder.newSetBinder(binder, GuiceProbe.class)
+            .addBinding().to(DomainAdminProbe.class));
+
+    @BeforeEach
+    void setUp(TwakeCalendarGuiceServer server) {
+        PreemptiveBasicAuthScheme basicAuthScheme = new PreemptiveBasicAuthScheme();
+        basicAuthScheme.setUserName(USERNAME.asString());
+        basicAuthScheme.setPassword(PASSWORD);
+        RestAssured.defaultParser = Parser.JSON;
+        RestAssured.requestSpecification = new RequestSpecBuilder()
+            .setContentType(JSON)
+            .setAccept(JSON)
+            .setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(StandardCharsets.UTF_8)))
+            .setPort(server.getProbe(RestApiServerProbe.class).getPort().getValue())
+            .setBasePath("")
+            .setAuth(basicAuthScheme)
+            .build();
+
+        server.getProbe(CalendarDataProbe.class)
+            .addDomain(DOMAIN);
+
+        server.getProbe(CalendarDataProbe.class)
+            .addUser(USERNAME, PASSWORD);
```

Comment:
Syntactic sugar refactoring idea:

```suggestion
        server.getProbe(CalendarDataProbe.class)
            .addDomain(DOMAIN)
            .addUser(USERNAME, PASSWORD);
```